In [26]:
import pandas as pd
label_df = pd.read_csv('../cleaned_dataset/cleaned_label_loc.csv')

In [27]:
label_df.head()

started_at                finished_at           room floor
0  2023-04-10 14:21:46+09:00  2023-04-10 14:21:50+09:00        kitchen   5th
1  2023-04-10 14:21:55+09:00  2023-04-10 14:22:16+09:00      cafeteria   5th
2  2023-04-10 14:22:23+09:00  2023-04-10 14:23:25+09:00        kitchen   5th
3  2023-04-10 14:23:29+09:00  2023-04-10 14:23:44+09:00       cleaning   5th
4  2023-04-10 14:23:42+09:00  2023-04-10 14:24:05+09:00  nurse station   5th

In [28]:
all_rooms = list(label_df['room'].unique())
print(sorted(all_rooms))

['501', '502', '503', '505', '506', '508', '510', '511', '512', '513', '515', '516', '517', '518', '520', '521', '522', '523', 'cafeteria', 'cleaning', 'hallway', 'kitchen', 'nurse station']


### Note: room 507 never appear in training data

In [36]:
ble_df = pd.read_csv('../cleaned_dataset/labelled_ble_data.csv')
all_beacon = list(ble_df['mac address'].unique())

In [37]:
print(sorted(all_beacon))

[np.int64(1), np.int64(2), np.int64(3), np.int64(4), np.int64(5), np.int64(6), np.int64(7), np.int64(8), np.int64(9), np.int64(10), np.int64(11), np.int64(12), np.int64(13), np.int64(14), np.int64(15), np.int64(16), np.int64(17), np.int64(18), np.int64(19), np.int64(20), np.int64(21), np.int64(22), np.int64(23)]


In [38]:
for i in range (1, 26):
    if i not in all_beacon:
        print(i)

24
25


### Note: in the final csv data we use for creating model, beacon 24 and 25 never appear

In [40]:
# now check again the rooms in labelled data we have 
all_rooms_labelled_ble_data = list(ble_df['room'].unique())
print(sorted(all_rooms_labelled_ble_data))

['501', '502', '503', '505', '506', '508', '510', '511', '512', '513', '515', '516', '517', '518', '520', '522', '523', 'cafeteria', 'cleaning', 'hallway', 'kitchen', 'nurse station']


In [43]:
for room in all_rooms: 
    if room not in all_rooms_labelled_ble_data:
        print(room)

521


### So we can see that in our final version of csv data for model building: we do not have 2 rooms: 507 and 521, and also the 2 beacons 24 and 25 never appear

In [46]:
# Now we would check the distribution of beacon detection among each rooms
for room in all_rooms_labelled_ble_data:
    temp_df = ble_df[ble_df['room'] == room]
    print("=" * 80)
    print("Here is the value count for room " + room)
    print(temp_df['mac address'].value_counts())
    print("\n")

Here is the value count for room kitchen
mac address
14    88108
4     56767
6      8027
9      4180
7      1397
5      1362
18     1243
19     1158
20      828
8       732
10      668
21      583
22      445
11      390
23      353
12      153
17      110
3        35
2        18
16       12
15        6
13        6
Name: count, dtype: int64


Here is the value count for room cafeteria
mac address
4     64803
14    56999
9     26420
6     10957
7      9059
19     8838
18     7130
20     3709
8      3543
10     1854
21     1767
5      1690
11     1645
22      840
23      538
12      160
17      106
3       105
13      104
16       91
15       51
2        24
Name: count, dtype: int64


Here is the value count for room cleaning
mac address
19    7626
4     3412
9     3367
18     835
5      656
6      576
7      527
20     410
8      314
17     202
10      55
11      41
22      36
3       24
16      24
14      12
21       6
Name: count, dtype: int64


Here is the value count for room nurse 